### PR TITLE
Set up Gradle doctor

### PR DESCRIPTION
**Pull request by bmarty**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [x] Technical
- [ ] Other :

## Content

Gradle Doctor is a gradle plugin that can help to detect configuration issue or other problems, and give "prescription" to improve the codebase and/or developer's environment.

Let's see how the CI accept this change.

Locally it already fixes an issue with missing JAVA_HOME key in the environment, which allows to have faster build from AS and command line.

https://runningcode.github.io/gradle-doctor/

## Motivation and context

Add a tool to help developers.

## Screenshots / GIFs

NA

## Tests

NA

## Tested devices

NA

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

**Comment by bmarty** [comment]
OK, so CI is a bit too sick. Maybe worth disabling gradle doctor on the CI?

**Comment by ouchadam** [comment]
:+1: for creating the tools script directory, could be nice to move the coverage file there as well in the future

File: tools/gradle/doctor.gradle [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+         */
+        ensureJavaHomeMatches = true
+        /**
+         * Ensure we have JAVA_HOME set.
+         */
+        ensureJavaHomeIsSet = true
+        /**
+         * Fail on any `JAVA_HOME` issues.
+         */
+        failOnError.set(!isCiBuild)
```
Comment (ouchadam) [inline tools/gradle/doctor.gradle:85]: should we fail for all users?

**Review by ouchadam — APPROVED** [review]

File: tools/gradle/doctor.gradle [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        /**
+         * Extra message text, if any, to show with the Gradle Doctor message. This is useful if you have a wiki page or
+         * other instructions that you want to link for developers on your team if they encounter an issue.
+         */
+        extraMessage.set("Here's an extra message to show.")
```
Comment (onurays) [inline tools/gradle/doctor.gradle:90]: We can make this useful later.

**Review by onurays — APPROVED** [review]
Nice plugin! Let's experience how it will behave with this configuration.

File: tools/gradle/doctor.gradle (hunk 1 above)
Comment (bmarty) [reply tools/gradle/doctor.gradle:85]: Yes. I have just copied the default configuration for now.

**Pull request merged**